pkg/apis/nsx.vmware.com/v1alpha2/ippool_types.go

```diff
@@ -0,0 +1,81 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha1"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:storageversion
+
+// IPPool is the Schema for the ippools API.
+// +kubebuilder:printcolumn:name="Type",type=string,JSONPath=`.spec.type`,description="Type of IPPool"
+// +kubebuilder:printcolumn:name="Subnets",type=string,JSONPath=`.status.subnets[*].cidr`,description="CIDRs for the Subnet"
+type IPPool struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   IPPoolSpec   `json:"spec"`
+	Status IPPoolStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// IPPoolList contains a list of IPPool.
+type IPPoolList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []IPPool `json:"items"`
+}
+
+// IPPoolSpec defines the desired state of IPPool.
+type IPPoolSpec struct {
+	// Type defines the type of this IPPool, public or private.
+	// +kubebuilder:validation:Enum=public;private
+	// +kubebuilder:default=private
+	Type string `json:"type"`
```

Comment:
should we add json:"string,omitempty"` ?

Reply:
There is no need, if no Type is defined, then use the default value and ippool_controller would use the default value to check whether we should get the type from vpc network config.

Reply:
I assume you are saying there is no impact for the ippool controller logic in nsx operator, right.

But for other client that wants to create an ippool CR, If no omitempty supported on Type, the client always has to provide Type field with a valid value (Public or Private) when building ippool spec that will be sent to apiserver later.

cc @dantingl  for comments.